src/main/java/com/palantir/docker/compose/execution/DefaultDockerCompose.java

```diff
@@ -56,50 +53,51 @@ public DefaultDockerCompose(DockerComposeFiles dockerComposeFiles, DockerMachine
     }
 
     public DefaultDockerCompose(DockerComposeExecutable rawExecutable, DockerMachine dockerMachine) {
+        super("docker-compose", new SynchronousDockerComposeExecutable(rawExecutable, log::debug));
```

Comment:
My intention with the suggestions above is that this would now become something like:

    public DefaultDockerCompose(DockerComposeExecutable rawExecutable, DockerMachine dockerMachine) {
        this.command = new Command(rawExecutable, log::debug);
        ...
    }

    public void build() {
        command.execute(throwingOnError(), "build");
    }

    ...

Doing this means this class and Docker would no longer share a common parent which matches the actual methods and behaviour on the classes - they both execute a command underneath and need to handle errors etc but that is abstracted away from the consumer of this class. This would avoid somebody using AbstractCommand as a type and then ending up with an object that must be cast to something else before any method can be invoked which is a pattern that I would like to avoid due to the difficultly that arises when adding new polymorphic types that the casting no longer accounts for.